### DISCUSSION TITLE
"out of gas" error when using view method

### DISCUSSION THREAD

**Discussion by escottalexander** [NomicFoundation/hardhat#4514] [Troubleshooting]
I have a view method defined in my contract that contains for loops and calls to other contract view methods. When I use Hardhat console to call this method it returns the following error:

```
error: ProviderError: Transaction ran out of gas
      at HttpProvider.request (/home/elliott/dev/generate-svg-contract-structure/packages/hardhat/node_modules/hardhat/src/internal/core/providers/http.ts:88:21)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async EthersProviderWrapper.send (/home/elliott/dev/generate-svg-contract-structure/packages/hardhat/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```
If I send fewer parameters in the method then it works as expected. I was not aware of any gas limitation when calling a view method. Any ideas why I am getting this error?

This is how I am calling the method in Hardhat console `await theContract.renderMain([0,1,2,3,4])`
I have also tried `await theContract.callStatic.renderMain([0,1,2,3,4])` and it yeilds the same error.

Here is the method that is being called: 
```
function renderMain(
        uint8[] memory layers
	) public view returns (string memory) {
        string memory result;
        for (uint8 i; i < layers.length; i++) {
            address[] memory locations = assets[layers[i]];
            bytes4[] memory colors = assetColors[layers[i]];
            for (uint h; h < locations.length; h++) {
                result = string(abi.encodePacked(result, IRenderContract(locations[h]).renderMain(colors)));
            }
        }
		return result;
	}
```

**Comment by fvictorio** [comment]
I think this can happen if the execution uses more gas than the block gas limit. Try increasing it in the Hardhat network config.
